queries/html.scm

```diff
@@ -0,0 +1,93 @@
+;;!! <aaa>
+;;!   ^^^
+;;!  -----
+(start_tag
+  (tag_name) @name
+) @_.domain
+
+;;!! </aaa>
+;;!    ^^^
+;;!  ------
+(end_tag
+  (tag_name) @name
+) @_.domain
+
+;;!! <aaa id="me">
+;;!       ^^^^^^^
+(attribute) @attribute
+
+;;!! <aaa id="me">
+;;!       ^^
+(attribute
+  (attribute_name) @collectionKey @collectionKey.trailing.start.endOf
+  [
+    (quoted_attribute_value)
+    (attribute_value)
+  ] ? @collectionKey.trailing.end.startOf
+) @_.domain
+
+;;!! <aaa value=2>
+;;!             ^
+;;!! <aaa id="me">
+;;!          ^^^^
+(attribute
+  (attribute_name) @value.leading.start.endOf
+  [
+    (quoted_attribute_value)
+    (attribute_value)
+  ] @value @value.leading.end.startOf
+) @value.domain
+
+;;!! <aaa id="me">
+;;!           ^^
+(quoted_attribute_value
+  (attribute_value) @textFragment
+) @string
+
+;;!! <aaa>
+;;!  ^^^^^
+(start_tag) @attribute.iteration @collectionKey.iteration @value.iteration
+
+;;!! <!-- comment -->
+;;!  ^^^^^^^^^^^^^^^^
+(comment) @comment @textFragment
+
+;;!! <aaa>text</aaa>
+;;!       ^^^^
+(text) @textFragment
+
+;;!! <script>hello</script>
+;;!          ^^^^^
+(raw_text) @textFragment
+
+;; Use parent wildcard to get all three kinds of elements: element, script_element, style_element
+
+;;!! <aaa>text</aaa>
+;;!  ^^^^^^^^^^^^^^^
+;;!       ^^^^
+(_
+  (start_tag) @xmlElement.interior.start.endOf
+  (end_tag) @xmlElement.interior.end.startOf
+) @xmlElement
+
+;;!! <aaa>text</aaa>
+;;!  ^^^^^    ^^^^^^
+;;!  ---------------
+(_
+  (start_tag) @xmlStartTag
+  (end_tag) @xmlEndTag
+) @_.domain
+
+(_
+  [
+    (start_tag)
+    (end_tag)
+  ] @xmlBothTags
+  (#allow-multiple! @xmlBothTags)
+) @_.domain
+
+(_
+  (start_tag) @xmlElement.iteration.start.endOf @xmlBothTags.iteration.start.endOf
+  (element)
```

Comment:
why do we require a child element? we don't do that for jsx

Reply:
Just because if you are in the middle off a text only element I don't think wish used that as iteration scope. In this example if the cursor is at the pipe and you say take every element I want both of them to be selected. And I think we also have tests for that since before that broke originally.
```html
<a>hello|stuff</a>
<b>something</b>
```